doc/release/3.4.0.rst

```diff
@@ -0,0 +1,249 @@
+Peak performance is achieved when scanning embedded field types.
+
+Querying full tuples, like in memtx, is also supported, but the performance is
+worse compared to memtx, because a tuple has to be constructed on the runtime
+arena from individual field values gathered from each column tree.
+
+Other features include:
+* Point lookup.
+* Stable iterators.
+* Insert/replace/delete/update.
+* Batch insertion in the Arrow format.
+* Transactions, including cross-engine transactions with memtx
+  (with ``memtx_use_mvcc_engine = false``).
+* Read view support.
+* Secondary indexes with an ability to specify covered columns and sequentially scan
+  indexed + covered columns.
+
+Embedded field types include only fixed-width types:
+* Integer: (u)int8/16/32/64.
+* Floating point: float32/64.
+
+Types with external storage include:
+* Strings.
+* All the other types supported by Tarantool: UUID, Decimal, Datetime, etc.
+
+By default, NULL values are stored explicitly and use up the same space as
+any other valid column value (1, 2, 4 or 8 bytes depending on an exact field
+type), however RLE encoding of NULLs is also supported. For reference,
+RLE-encoding of a column with 90% evenly distributed NULL values reduces
+memory consumption of that column by around 5 times.
+
+.. _3-4-cross-engine:
+
+[CE] Memtx <-> vinyl cross-engine transactions
+----------------------------------------------
+
+Tarantool now supports mixing statements for memtx and vinyl in the same transaction,
+for example:
+
+.. code-block:: lua
+
+    local memtx = box.schema.space.create('memtx', {engine = 'memtx'})
+    memtx:create_index('primary')
+    local vinyl = box.schema.space.create('vinyl', {engine = 'vinyl'})
+    vinyl:create_index('primary')
+
+    memtx:insert({1, 'a'})
+    vinyl:insert({2, 'b'})
+
+    box.begin()
+    memtx:replace(vinyl:get(2))
+    vinyl:replace(memtx:get(1))
+    box.commit()
+
+.. note::
+
+   * Accessing a vinyl space may trigger a fiber yield (to read a file from the disk),
+     so MVCC must be enabled in memtx to make use of the new feature:
+
+     .. code-block:: lua
+
+         box.cfg{memtx_use_mvcc_engine = true}
+
+   * Vinyl operations may yield implicitly, so a transaction may be aborted
+     with TRANSACTION_CONFLICT in case of concurrent transactions.
+
+.. _3-4-native:
+
+[EE] New boostrap strategy in failover: ``native``
+--------------------------------------------------
+
+Now supervised failover coordinator supports three bootstrap strategies:
+native, supervised, auto.
+
+The new ``native `` strategy acts more or less similar to the ``auto`` strategy,
+but relaxes its limitations. It is based on the supervised strategy and basically
+performs two things:
+* issues ``box.ctl.make_bootstrap_leader({graceful = true})`` to bootstrap
+  a replicaset;
+* issues ``box.ctl.make_bootstrap_leader()`` to keep the bootstrap leader
+  record pointing to the instance that is currently in the RW mode (to register
+  new replicas).
```

Comment:
I'm not sure that the calls under the hood are really interesting for our reader. Maybe some motivational things would work better here.

Quoted from my mini-announce in a team chat (in Russian):

> Очень похоже на дефолтную auto, но под капотом работает по-другому (поверх стратегии supervised). Цель была в том, чтобы решить следующие проблемы:
>
> * Чтобы не стреляла ошибка Some replica set members were not specified in box.cfg.replication при одновременном подключении нескольких реплик, при наличии неанонимных CDC в репликасете, при наличии старых реплик в _cluster.
> * Чтобы база бутстрапилась по команде координатора, а не инстансами самостоятельно <...>.

Reply:
@Totktonada Please approve this text:

```
The new ``native`` strategy relaxes the limitations of the ``auto`` strategy,
but has different under-the-hood implementation (based on the ``supervised`` strategy).
Otherwise, it acts similar to the ``auto`` strategy.

In effect, it helps resolve these two problems:
* Avoid the error ``Some replica set members were not specified in box.cfg.replication``
  in the following cases:
  * several replicas join at the same time,
  * the replica set includes non-anonymous CDC instances,
  * ``_cluster`` contains expelled replicas.
* Make the database get bootstrapped upon the coordinator's command rather than
  let the instances boostrap it on their own.
```